packages/grid/x-data-grid-premium/src/DataGridPremium/useDataGridPremiumComponent.tsx

```diff
@@ -145,7 +145,7 @@ export const useDataGridPremiumComponent = (
   useGridRowPinning(privateApiRef, props);
```

Comment:
> NOTE: The results above don't use the same set of 100,000 rows (it's the Employee dataset, randomly generated), thus why the Count differs.

Not sure if that would be useful, but you can disable random generation (we do this for screenshot generation):
```diff
diff --git a/packages/grid/x-data-grid-generator/src/services/random-generator.ts b/packages/grid/x-data-grid-generator/src/services/random-generator.ts
index a379ef281..77529aa15 100644
--- a/packages/grid/x-data-grid-generator/src/services/random-generator.ts
+++ b/packages/grid/x-data-grid-generator/src/services/random-generator.ts
@@ -15,7 +15,7 @@ import { GridDataGeneratorContext } from './gridColDefGenerator';
 const chanceId = globalChance();
 let chance: Chance.Chance;
 
-declare const DISABLE_CHANCE_RANDOM: any;
+const DISABLE_CHANCE_RANDOM = true;
 
 if (typeof DISABLE_CHANCE_RANDOM !== 'undefined' && DISABLE_CHANCE_RANDOM) {
   chance = globalChance(() => 0.5);
```

<img width="829" src="https://github.com/mui/mui-x/assets/13808724/29874686-5380-440b-85ae-d9c37aa67414">

Reply:
Good to know, thanks!